### PR TITLE
Add Darwin as platform to various "broken" packages that actually build

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change

Adds Darwin as platform to various "broken" packages that actually build.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by matthewbauer** [comment]
@LnL7 got any advice for this? I could definitely split this up if it's too much. They're all good changes though and I hate clogging up the pr with 30+ minor fixes.

File: pkgs/development/libraries/opencv/default.nix [hunk 1]
```diff
@@ -81,6 +83,6 @@ stdenv.mkDerivation rec {
     homepage = http://opencv.org/;
     license = stdenv.lib.licenses.bsd3;
     maintainers = with stdenv.lib.maintainers; [viric flosse];
-    platforms = with stdenv.lib.platforms; linux;
+    platforms = with stdenv.lib.platforms; linux ++ darwin;
```
Comment (LnL7) [inline pkgs/development/libraries/opencv/default.nix]: I'm not sure what's going on here, but it seems like the `darwin` variable here is shadowed by the function argument for some reason.

```
error: value is a set while a list was expected, at pkgs/development/libraries/opencv/default.nix:86:44
```

**Comment by matthewbauer** [comment]
Tesseract, wordnet, anti word and libdvdread will probably have the same issue.

**Comment by LnL7** [comment]
I only checked the packages with actual changes, I'll add everything to the list to make sure.
I think it's mainly personal preference, but I think bringing the licenses or platforms into scope is usually not really worth it since those are only used once or twice.

**Comment by LnL7** [comment]
Can you fix these?

```
in job ‘pkgs.vte’:
value is null while a set was expected, at [1m/nix/store/ccdxc5241h8mqj06d761rmkxxabny9hb-git-export/lib/attrsets.nix[0m:199:76

in job ‘pkgs.corn’:
value is null while a set was expected, at [1m/nix/store/ccdxc5241h8mqj06d761rmkxxabny9hb-git-export/lib/attrsets.nix[0m:199:76

in job ‘pkgs.dconf’:
value is null while a set was expected, at [1m/nix/store/ccdxc5241h8mqj06d761rmkxxabny9hb-git-export/lib/attrsets.nix[0m:199:76

in job ‘pkgs.tessaract’:
value is null while a set was expected, at [1m/nix/store/ccdxc5241h8mqj06d761rmkxxabny9hb-git-export/lib/attrsets.nix[0m:199:76

in job ‘pkgs.fbida.x86_64-darwin’:
Package ‘sane-backends-1.0.25’ in ‘/nix/store/ccdxc5241h8mqj06d761rmkxxabny9hb-git-export/pkgs/applications/graphics/sane/backends/generic.nix:62’ is not supported on ‘x86_64-darwin’, refusing to evaluate.
```

**Comment by matthewbauer** [comment]
Yeah I can get them. Interestingly, "miniupnpc" was already doing the same thing: https://github.com/ndowens/nixpkgs/blob/9d30abf7b201b6144883815fccf904e83555334b/pkgs/tools/networking/miniupnpc/default.nix#L21

I wonder if that was causing an error?

**Comment by LnL7** [comment]
It's only a problem for expressions that use [`pkgs.darwin`](https://github.com/NixOS/nixpkgs/pull/25066/commits/f9aca8a7504cc49386dcd32d3ff334f62fc36e3b#diff-3cbdb7c584894a24d039e5652877554bR13) as an argument. That's why it usually works.

**Comment by matthewbauer** [comment]
Ok so a rundown on those 3:

- fbida doesn't seem to ever have worked. I'm not sure why I had it in there. I just reverted it.
- pkgs.corn should be pkgs.cron, I messed up the name in the commit message, I can rework it if you want but I've been trying not to rebase unless absolutely necessary
- pkgs.vte should be pkgs.gnome3.vte

I can also squash the 2 reverted commits if you want but again I'm trying to avoid modifying commit hashes just in case you were working off of them.

**Comment by LnL7** [comment]
Looks good now, you can rebase/squash.

ps. having a list of attribute paths for a batch pr like this would be pretty useful for me, I have some hydra configuration that I usually use to build a set of packages for a pr.

**Comment by matthewbauer** [comment]
Ok everything looks good!

**Review by LnL7 — APPROVED** [review]

**Pull request merged**